### DISCUSSION TITLE
Holidata

### DISCUSSION THREAD

**Discussion by dstark** [GothenburgBitFactory/taskwarrior#3667] [Q&A] [answered]
What's the best way to have Taskwarrior display data from holidata.net? When I download one of the files (XML, JSON, YAML), and specify `include [pathtofile]/2024.[extension]` in `.taskrc`, Taskwarrior complains that the entry is malformed. I do see that those files are quite different from the default format (e.g., `holidays.en-US.rc`). But would anyone see what I'm missing for how to get Taskwarrior to ingest holidata properly?

Thanks so much!

**Comment by djmitche** [comment]
@lauft any ideas here?

**Comment by lauft** [comment]
In `doc/rc/` there is the `refresh` script, which in turn uses `scripts/add-ons/update-holidays.pl` to download an convert the locale from JSON to Taskwarrior's rc format.

I am not that familiar with Taskwarrior's script setup though (Timewarrior has a Python script for this), so there may be some tweaking involved...

**Reply by dstark** [reply]
@lauft and @djmitche, thanks so much for your thoughts here! I'm admittedly coming from a very newbie perspective. So, I'm having trouble finding `doc/rc/refresh` to try to run it (on Ubuntu via the Windows Subsystem for Linux). Could I ask where I should be seeing this?

Just to clarify, I currently only have Taskwarrior installed, not Timewarrior also.

Thanks, again, for your guidance!

**Comment by lauft** [comment]
> Could I ask where I should be seeing this?

According to CMake configuration, it should be installed in a directory `share/doc/task/rc`, so try looking in `/usr/local/share/doc/task/rc` for the refresh script. As a last resort a `find / -name 'refresh'` could help 🤞🏻

**Reply by dstark** [reply]
> > Could I ask where I should be seeing this?
> 
> According to CMake configuration, it should be installed in a directory `share/doc/task/rc`, so try looking in `/usr/local/share/doc/task/rc` for the refresh script. As a last resort a `find / -name 'refresh'` could help 🤞🏻

Thanks so much for these suggestions, @lauft. `find / -name 'refresh'` returns no positive results—just a string of "permission denied" errors for different folders until it finally terminates with the error `File system loop detected; ‘/mnt/wslg/distro’ is part of the same file system loop as ‘/’.` And for me `/usr/local/share` contains only ca-certificates, fonts, man, sgml, xml.

There is `/usr/share/doc`, and this contains a `taskwarrior` folder (but no `task`, `rc`, or `refresh`), and that `taskwarrior` folder contains only NEWS.gz, README.Debian, README.md.gz, TODO.Debian, changelog.Debian.gz, copyright, examples.

Combined with the `find` results, does this suggest that I don't actually have `refresh` installed with my Taskwarrior currently (although or because it's 2.6.2 from the Ubuntu repository)?

**Comment by lauft** [comment, marked as answer]
> Combined with the find results, does this suggest that I don't actually have refresh installed with my Taskwarrior currently (although or because it's 2.6.2 from the Ubuntu repository)?

This might be the case. 🤔  If those scripts are not included in the distribution, one will have to contact the maintainers to fix this.

Until then you can try to download the [update-holidays.pl](https://github.com/GothenburgBitFactory/taskwarrior/blob/develop/scripts/add-ons/update-holidays.pl) script (requires Perl) and execute it
```
update-holidays.pl --locale $locale --file holidays.${locale}.rc 
```

**Reply by dstark** [reply]
Got it. Thanks so much!
